### PR TITLE
Table

### PR DESCRIPTION
**Pull request by SergeyOdnosteblets**

### CODE REVIEW

**Review by t-shevchuk — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -87,13 +87,24 @@ export const App: React.FC = () => {
...
         <Routes>
-          <Route path="/" element={<ListContainer list={list} setList={setList} />} />
+          <Route
+            path="/"
+            element={
+              <ListContainer
+                list={list}
+                setList={setList}
+                setSortUsers={setSortUsers}
+                sortUsers={sortUsers}
```
Comment (t-shevchuk) [inline src/App.tsx]: naming for both

File: src/components/Filter/Filter.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ export const Filter: React.FC<FilterTypes> = ({setFilteredUsers, onSubmit }) =>
   return (
     <div>
       <div className={styles.form} onClick={(e) => e.stopPropagation()}>
-        <form onSubmit={handleSubmit(onSubmit)}>
+        <form onSubmit={handleSubmit(onSubmit) } >
```
Comment (t-shevchuk) [inline src/components/Filter/Filter.tsx]: formatting

File: src/components/List/List.module.scss.d.ts [hunk 3]
```diff
@@ -3,3 +3,5 @@ export const user: string;
 export const item: string;
 export const button:string
 export const item__buttons: string
+export const table:string
```
Comment (t-shevchuk) [inline src/components/List/List.module.scss.d.ts]: format this file

File: src/components/List/List.tsx [hunk 4]
```diff
@@ -24,26 +27,50 @@ export const List: React.FC<UserListItemProps> = ({
     });
   };
 
+  const handleTableFilter = (id: string | number) => {
+    setSortUsers(id);
+  };
+
+  useEffect(() => {
+    if (sortUsers) {
```
Comment (t-shevchuk) [inline src/components/List/List.tsx]: not obvious naming

File: src/components/List/List.tsx [hunk 5]
```diff
@@ -24,26 +27,50 @@ export const List: React.FC<UserListItemProps> = ({
     });
   };
 
+  const handleTableFilter = (id: string | number) => {
```
Comment (t-shevchuk) [inline src/components/List/List.tsx]: You don't handle filtering here

File: src/pages/ListContainer/ListContainer.tsx [hunk 6]
```diff
@@ -46,7 +49,14 @@ export const ListContainer: React.FC<ListTypes> = ({ list, setList }) => {
         <button className={styles.button} onClick={() => handleEdit()}>
           Add User
         </button>
-        <List removeUser={removeUser} handleEdit={handleEdit} firlteredUsers={firlteredUsers} />
+        <List
+          removeUser={removeUser}
+          handleEdit={handleEdit}
+          firlteredUsers={firlteredUsers}
```
Comment (t-shevchuk) [inline src/pages/ListContainer/ListContainer.tsx]: typo `filtered`

File: src/pages/ListContainer/ListContainer.tsx [hunk 7]
```diff
@@ -57,6 +67,34 @@ export const ListContainer: React.FC<ListTypes> = ({ list, setList }) => {
           />
         )}
       </div>
+      <div>
+        <CSVReader
+          onUploadAccepted={(results: any) => {
+            if (results.errors.length) {
+              alert('faildToAddCount');
```
Comment (t-shevchuk) [inline src/pages/ListContainer/ListContainer.tsx]: typo `failed`

File: src/pages/ListContainer/ListContainer.tsx [hunk 8]
```diff
@@ -57,6 +67,34 @@ export const ListContainer: React.FC<ListTypes> = ({ list, setList }) => {
...
+              alert('addedCount');
+            }
+          }}>
+          {({ getRootProps }: any) => (
+            <button type="button" {...getRootProps()} className={styles.button}>
+              Import Contacts
+            </button>
+          )}
+        </CSVReader>
+      </div>
```
Comment (t-shevchuk) [inline src/pages/ListContainer/ListContainer.tsx]: Why in this PR are old changes? that should be merged and new branch should have the last sources

**Review by t-shevchuk — CHANGES_REQUESTED** [review]

File: src/components/List/List.tsx [hunk 9]
```diff
@@ -27,47 +28,50 @@ export const List: React.FC<UserListItemProps> = ({
     });
   };
 
-  const addSortCategory = (category: string | number) => {
-    setSortedUsers(category);
+  const sortBy = (category: string | number) => {
```
Comment (t-shevchuk) [inline src/components/List/List.tsx]: onSortingChange

File: src/App.tsx [hunk 10]
```diff
@@ -87,7 +88,9 @@ export const App: React.FC = () => {
       country: 'United States',
     },
   ]);
-  const [sortedUsers, setSortedUsers] = useState<string | number>('');
+  const [sortCategory, setSortCategory] = useState<string | number>('');
+  const [filteredUsers, setFilteredUsers] = useState<UserInfo[]>(list);
```
Comment (t-shevchuk) [inline src/App.tsx:91]: Why do you need it here?

File: src/App.tsx [hunk 11]
```diff
@@ -87,7 +88,9 @@ export const App: React.FC = () => {
...
+  const [forceRefresh, setForceRefresh] = useState(true);
```
Comment (t-shevchuk) [inline src/App.tsx]: why?

File: src/components/List/List.tsx [hunk 12]
```diff
@@ -11,8 +11,9 @@ export const List: React.FC<UserListItemProps> = ({
   handleEdit,
   filteredUsers,
   setFilteredUsers,
-  sortedUsers,
-  setSortedUsers,
+  sortCategory,
```
Comment (t-shevchuk) [inline src/components/List/List.tsx]: sortBy, setSortBy

File: src/components/List/List.tsx [hunk 13]
```diff
@@ -27,47 +28,50 @@ export const List: React.FC<UserListItemProps> = ({
...
 
   useEffect(() => {
-    if (sortedUsers) {
-      const sorted = [...filteredUsers].sort((a, b) => (a[sortedUsers] > b[sortedUsers] ? 1 : -1));
+    if (sortCategory) {
+      const sorted = [...filteredUsers].sort((a, b) =>
+        a[sortCategory].toLowerCase() > b[sortCategory].toLowerCase() ? 1 : -1,
+      );
       setFilteredUsers(sorted);
+      console.log('rerender');
```
Comment (t-shevchuk) [inline src/components/List/List.tsx]: remove

File: src/types/ListTypes.ts [hunk 14]
```diff
@@ -2,7 +2,10 @@ import { UserInfo } from "./UserInfo";
 
 export interface ListTypes {
     list: UserInfo[]
-    setList: (list: UserInfo[]) => void ; 
-    sortedUsers: string | number
-    setSortedUsers: React.Dispatch<React.SetStateAction<string | number>>
+    sortCategory: string | number
+    setSortCategory: React.Dispatch<React.SetStateAction<string | number>>
```
Comment (t-shevchuk) [inline src/types/ListTypes.ts]: you don't use dispatch in your project. that's just a function

File: src/types/UserListItemProps.ts [hunk 15]
```diff
@@ -3,9 +3,10 @@ import { UserInfo } from "./UserInfo";
 export interface UserListItemProps {
     removeUser: (userObj: UserInfo) => void 
     handleEdit: (userObj: UserInfo) => void
-    sortedUsers: string | number
-    setSortedUsers: React.Dispatch<React.SetStateAction<string | number>>
+    sortCategory: string | number
+    setSortCategory: React.Dispatch<React.SetStateAction<string | number>>
```
Comment (t-shevchuk) [inline src/types/UserListItemProps.ts]: same here

File: src/types/UserToEdit.ts [hunk 16]
```diff
@@ -4,7 +4,9 @@ export interface UserToEdit {
     userToEdit?: UserInfo | null
     setIsModalActive: React.Dispatch<React.SetStateAction<boolean>>
```
Comment (t-shevchuk) [inline src/types/UserToEdit.ts]: and here

File: src/App.tsx (hunk 10 above)
Comment (SergeyOdnosteblets) [reply src/App.tsx:91]: changed

File: src/App.tsx (hunk 11 above)
Comment (SergeyOdnosteblets) [reply src/App.tsx]: changed

File: src/components/Filter/Filter.tsx [hunk 17]
```diff
@@ -20,7 +20,7 @@ export const Filter: React.FC<FilterTypes> = ({setFilteredUsers, onSubmit }) =>
   return (
     <div>
       <div className={styles.form} onClick={(e) => e.stopPropagation()}>
-        <form onSubmit={handleSubmit(onSubmit)}>
+        <form onSubmit={handleSubmit(onSubmit)} >
```
Comment (t-shevchuk) [inline src/components/Filter/Filter.tsx]: formatting

File: src/components/List/List.module.scss [hunk 18]
```diff
@@ -28,3 +28,21 @@
...
+.table__body {
+  background: rgb(164, 91, 91);
+  text-align: center;
+  font-weight: bold;
+  color: white;
+
+  tr {
+    border: 2px solid black;
+  }
+
```
Comment (t-shevchuk) [inline src/components/List/List.module.scss]: formatting

**Pull request merged**